File: Shruti/WFTextEditor/BaseBillingWithOffer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WFTextEditor
{
    public abstract class BaseBillingWithOffer : BaseBilling
    {
        protected double discountPercentage { get; set; }
        public abstract void applyDiscount(double discountPercentage);

        public override double CalculateBill(double cost, double tax)
        {
            return cost + (cost * tax) - (cost * discountPercentage);
        }

    }
}

File: NotePad_aishwarya/src/com/training/billing/BaseBillingWithOffer.java
package com.training.billing;

public abstract class BaseBillingWithOffer extends BaseBilling {

	protected double discountPercent;

	public void applyDiscount(double discountPercent) {
	}

	@Override
	public double calculateBill(double cost, double tax) {
		return cost + (cost * tax) - (cost * discountPercent);
	}
}
